### ISSUE TITLE
[Bug] Dash video requests don't reflect port settings

### ISSUE THREAD

**Issue by leana8959** [iv-org/invidious#4794]
**Describe the bug**
Having invidious behind a reverse proxy, I noticed that the Dash quality mode doesn't work.
HD720 works correctly.

I could see that the browser was sending requests to the wrong port (the proxied port 3001, not the external port 443). I have configured the external port to be 443, but that seem to have no effect on this behaviour.

<img width="424" alt="wrong port browser" src="https://github.com/user-attachments/assets/3551176f-ff66-4087-aa11-3955b416f631">

**Steps to Reproduce**

I use docker-compose on the plain example configuration (with port 3001) and run the container behind caddy.

<details>
<summary>caddy configuration</summary>
I route invidious to localhost:3001

```Caddyfile
https://myawesome.domain.com {
        respond /robots.txt 200 {
                body "User-agent: *
Disallow: /"
        }

        reverse_proxy localhost:3001
}
```

</details>

<details>
<summary>compose.yaml</summary>

TL;DR: I changed the ports to 3001

```yaml
volumes:
  invidious_postgres:
    external: true

services:
  invidious:
    image: quay.io/invidious/invidious:latest
    restart: unless-stopped
    ports:
      - "127.0.0.1:3001:3000"
    environment:
      INVIDIOUS_CONFIG: |
        db:
          dbname: invidious
          user: kemal
          password: kemal
          host: invidious_db
          port: 5432
        check_tables: true
        https_only: true
        domain: myawesome.domain.com
        external_port: 443
        registration_enabled: true
        hmac_key: "somekey"
    healthcheck:
      test: wget -nv --tries=1 --spider http://127.0.0.1:3000/api/v1/comments/jNQXAC9IVRw || exit 1
      interval: 30s
      timeout: 5s
      retries: 2
    depends_on:
      - invidious_db

  invidious_db:
    image: docker.io/library/postgres:14
    restart: unless-stopped
    volumes:
      - invidious_postgres:/var/lib/postgresql/data
      - ./config/sql:/config/sql
      - ./docker/init-invidious-db.sh:/docker-entrypoint-initdb.d/init-invidious-db.sh
    environment:
      POSTGRES_DB: invidious
      POSTGRES_USER: kemal
      POSTGRES_PASSWORD: kemal
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $$POSTGRES_USER -d $$POSTGRES_DB"]
```

</details>

**Screenshots**
"The media could not be load, either because the server or network failed or because the format is not supported."

Safari gives a more detailed error message, showing that the request is going to port 3001 and not 443.

- Firefox
<img width="914" alt="firefox error"
src="https://github.com/user-attachments/assets/a83d0399-6bd0-4b30-a809-6745df71fd3f">

- Safari
<img width="1731" alt="safari error" src="https://github.com/user-attachments/assets/162c5f91-fbcb-4e66-9573-0c430f8c12cb">

**Additional context**
The problem can be reproduced both on Firefox and Safari
- Browser (if applicable): Firefox 128.0, Safari 17.5

Thanks and have a nice day :)

**Comment by nbuechner** [comment]
I am affected by this too.

**Comment by unixfox** [comment]
Hello, this sounds like a configuration issue on your side.

As maintainers, we do not have a lot of free time so you should ask your question on our [matrix channel](https://matrix.to/#/#invidious:matrix.org) or [IRC](https://web.libera.chat/?channel=#invidious).

Closing as it's a configuration issue on your side.

**Comment by leana8959** [comment]
I have figured out the issue: I have confused the option `port` and `external_port`. Really sorry for the time it took to figure this out !

@nbuechner Maybe check if your `external_port` is configured correctly, I have it set to the reverse proxy's port, yet it should be 443.

**Comment by mrjay42** [comment]
> I have figured out the issue: I have confused the option `port` and `external_port`. Really sorry for the time it took to figure this out !
> 
> [@nbuechner](https://github.com/nbuechner) Maybe check if your `external_port` is configured correctly, I have it set to the reverse proxy's port, yet it should be 443.

Ok this is a long shot, but could you share your complete docker-compose.yml file, please? <3

I'm encountering the same issue, and I'm not sure how to solve it, even by reading again and again your message 🥺
